webapp/store/survey/hooks/index.js

```diff
@@ -32,3 +36,58 @@ export const useNodeDefsByUuids = (uuids) => Survey.getNodeDefsByUuids(uuids)(us
 
 // ==== Categories
 export const useCategoryByUuid = (uuid) => Survey.getCategoryByUuid(uuid)(useSurvey())
+
+const useSurveyItem = ({ type }) => {
+  const surveyId = useSurveyId()
+  const [state, setState] = useState([])
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const { data } = await axios.get(`/api/survey/${surveyId}/${type}`, {
+        params: { draft: true, validate: false },
+      })
+      setState(data.list)
+    }
+    fetchData()
+  }, [surveyId])
+
+  const getItemsById = useCallback(
```

Comment:
can this be renamed into something like getItemsIndexedByUuid or something like that?

Reply:
can by i used to put by Id because not always it is by Uuid, what do you prefer?